rskj-core/src/main/java/co/rsk/cli/CliArgs.java

```diff
@@ -138,5 +150,43 @@ private O getOptionByName(String optionName) {
                             () -> new NoSuchElementException(String.format("-%s is not a valid option", optionName))
                     );
         }
+
+        /**
+         * Parses a string argument in the format e.g <i>database.dir=/home/rsk/core<i/> to a map in the following
+         * structure:
+         * <blockquote>
+         *     {
+         *         "database": {
+         *             "dir": "/home/rsk/core"
+         *         }
+         *     }
+         * </blockquote>
+         * @param arg to parse
+         * @return a string arg parsed to an equivalent map having the same structure system properties would have.
+         */
+        private Map<String, Object> parseArgToMap(String arg) {
```

Comment:
wondering if we really need to parse it on our own?.. Check [this](https://github.com/rsksmart/rskj/blob/IRIS-3.1.0/rskj-core/src/main/java/co/rsk/config/NodeCliOptions.java#L41) out - seems the `typesafe` lib has its own capability to parse path-based properties

Reply:
Done